src/Server.cpp

```diff
@@ -852,33 +950,97 @@ void handleCompleteRequestHeader(Server& server, int clientFd, Connection& conne
 	}
 
 	server.findTargetResource(connection, connection.m_request);
+	if (isCGIRequested(connection)) {
+		connection.m_request.hasCGI = true;
+		ProcessOps processOps;
```

Comment:
I would register this as a private member const ProcessOps& while constructing the server.
Then you can just pass the same one (since we need to construct CGIHandler new everytime, you would also need a getter for the member).

This way you can construct the server with a MockProcessOps and it will the pass the Mock (dependency injection)

Reply:
Done